rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -309,7 +309,7 @@ final class ArrayType implements J, TypeTree, Expression {
         TypeTree elementType;
 
         @With
-        List<JRightPadded<Space>> dimensions;
+        List<AnnotatedDimension> dimensions;
```

Comment:
This change is not backwards compatible. While there probably aren't very many callers which need to be updated, there is also the compatibility of the serialization form to consider. While this could be addressed with a dedicated deserializer, we should still compare with alternatives, which are backwards compatible.

Reply:
Thanks for the inputs here, the backward compatibility of deserialization is also my concern. I'm not sure how to handle this correctly, do you think adding a separate field works? and mark the old field deprecated?

Reply:
I will try to take a look at it in the next few days.

Reply:
This came up again in https://github.com/openrewrite/rewrite/issues/3667 ; would it be possible to see this through?

Reply:
Yes, we really need to get this fixed...